transport.go

```diff
@@ -85,6 +85,9 @@ type Transport struct {
 	createdConn bool
 	isSingleUse bool // was created for a single server or client, i.e. by calling quic.Listen or quic.Dial
 
+	readingNonQUICPackets atomic.Bool
```

Comment:
Don't see the point of this bool. Why wouldn't we always have the buffer allocated (based on size param somewhere) and just drop on overflow (or when sized to zero).

Reply:
Filled with packet, this would add 32*1500 = ~48 kB of wasted memory. I'd like to avoid that.

Reply:
Not if the count is configurable and set to 0

Reply:
I was trying to avoid introduce yet more configuration API.

Reply:
It's only configurable if you need it. It's zero by default, so doesn't seem like you need to provide a value?

Reply:
There's still the race condition of calling this function before `Listen` or `Dial`. Then the channel hasn't been created yet.

Reply:
Not if it's initializes at point the transport is constructed? The channel can be nil by default at which point we'd just drop the packets?

Reply:
I'd still argue that this is not needed.

Reply:
I would still expose the responsibility of the channel creation/assignment to the builder of the transport type (before its used). Then you can use a nil check, and a user can control the buffer size (which is now quite small).